### PR TITLE
Enable custom paths for endpoints

### PR DESCRIPTION
**Pull request by Juliano1612**
# Description

This updates `ssx-core` interfaces, `ssx-sdk`, and `ssx-server` to allow for custom paths for endpoints on the client and the server.

# Type

- [x] New feature (non-breaking change which adds functionality)

# System Configuration

Node Version: 16.13.1
OS: ubuntu 20.04
Browser Version: Chrome 108.0.5359.94

# Diligence Checklist

- [x] This change requires a documentation update
- [x] I have included unit tests
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by skgbafa — APPROVED** [review]
Solid! Looks good to me

File: packages/ssx-core/src/client/types.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ export type SSXServerHost = string;
 /** The ssx-powered server configuration settings */
 export type SSXProviderServer = {
     host: SSXServerHost;
+    /** Optional configuration for the server's endpoints. */
+    endpoints?: SSXServerEndpoints;
```
Comment (w4ll3) [inline packages/ssx-core/src/client/types.ts]: can we call it routes?

File: packages/ssx-sdk/tests/ssx.test.ts [hunk 2]
```diff
@@ -36,6 +36,23 @@ test("Instantiate SSX with providers.web3.driver", () => {
   }).not.toThrowError();
 });
 
+test('Should override paths successfully', async () => {
```
Comment (w4ll3) [inline packages/ssx-sdk/tests/ssx.test.ts:39]: this test does not assure paths were overwritten

File: packages/ssx-server/tests/express.middleware.test.ts [hunk 3]
```diff
@@ -97,3 +97,15 @@ test('Should log out successfully', async () => {
 
   expect(res.statusCode).toEqual(204);
 });
+
+
+test('Should override paths successfully', async () => {
```
Comment (w4ll3) [inline packages/ssx-server/tests/express.middleware.test.ts]: also does not assure paths to be overwritten

File: packages/ssx-server/tests/http.middleware.test.ts [hunk 4]
```diff
@@ -81,3 +81,14 @@ test('Should log out successfully', async () => {
 
   expect(res.statusCode).toEqual(200);
 });
+
+test('Should override paths successfully', async () => {
```
Comment (w4ll3) [inline packages/ssx-server/tests/http.middleware.test.ts]: same

**Review by w4ll3 — CHANGES_REQUESTED** [review]

**Review by w4ll3 — APPROVED** [review]

**Pull request merged**